### ISSUE TITLE
Execution policy not accepting null access tokens

### ISSUE THREAD

**Issue by ericdunn** [nozzlegear/ShopifySharp#154]
Hello,

I'm calling the ShopifyService without using an access token (sending in null). This used to work fine until version 4.0. After updating all of my code to the new version, and then investigating this a little, I noticed that in the SmartRetryExecutionPolicy, when it tries to read the access token from the header, it's using :

```
      private string GetAccessToken(IFlurlClient client)
        {
            var headerValues = client.HttpClient.DefaultRequestHeaders.GetValues(REQUEST_HEADER_ACCESS_TOKEN);

            return headerValues.FirstOrDefault();
        }
```

The call to GetValues will actually fail if the key doesn't exist, so it will never make it to headerValues.FirstOrDefault().

An option to fix this would be:

```
        private string GetAccessToken(IFlurlClient client)
        {
            IEnumerable<string> values = new List<string>();
            return client.HttpClient.DefaultRequestHeaders.TryGetValues(REQUEST_HEADER_ACCESS_TOKEN, out values) ?
                values.FirstOrDefault() :
                null;
        }
```

The rest of the code properly deals with this value being null, so everything else works after.

Thanks,
Eric.

**Comment by nozzlegear** [comment]
Thanks for opening this issue, and for the quick fix! This should be fixed in v4.0.1 on Nuget.

**Comment by ericdunn** [comment]
Thanks for the quick fix and nuget update!
